File: LeetCode/844_BackspaceStringCompare.go
// https://leetcode.com/problems/backspace-string-compare/submissions/
// 2022年05月02日 12:50:10

func backspaceCompare(s string, t string) bool {
    // v1 
    nestedconvert:=func (s string) string {
        s1:=""
        for i:=0;i<len(s);i++{
            if s[i]=='#'{
                if len(s1)>0{
                    s1=s1[:len(s1)-1]
                }
            }else{
                s1+=string(s[i])
            }
        }
        return s1
    }
    return nestedconvert(s)==nestedconvert(t)
}

File: LeetCode_rename/844_BackspaceStringCompare.py
# https://leetcode.com/problems/backspace-string-compare/
# 2022年05月01日 21:42:03

class Solution:
    def backspaceCompare(self, s: str, t: str) -> bool:
        # v1 
        def convert(s):
            s1=''
            for i in range(len(s)):
                if s[i]=="#":
                    if len(s1)>0:
                        s1=s1[:-1]
                else:
                    s1+=s[i]
            return s1
        return convert(s)==convert(t)